DeusEx/Classes/DXRPasswords.uc

```diff
@@ -81,6 +81,21 @@ function RandoPasswords(int mode)
             ChangeATMPIN(a, i);
         }
     }
+
+    FixCodes();
+}
+
+function FixCodes()
+{
+    local string newpassword;
+    if( dxr.localURL == "02_NYC_HOTEL" ) {
+        newpassword = GeneratePasscode("4321");
```

Comment:
GeneratePasscode is "stable", as in if you give it the same seed and the same oldpassword it will always generate the same newpassword because of the way it calls SetSeed
`dxr.SetSeed( dxr.seed + dxr.Crc(oldpasscode) );`

so here we recalculate the newpassword for convenience, but then ReplacePassword adds it to the oldpasswords/newpasswords array with a hardcoded oldpassword using the text that actually appears in the datacube/note

Reply:
All things considered, this is a pretty clean solution